.circleci/config.yml

```diff
@@ -1401,6 +1419,40 @@ jobs:
           path: all-bytecode-reports.zip
       - gitter_notify_failure_unless_pr
 
+  c_release_binaries:
+    <<: *base_ubuntu2004
+    steps:
+      - checkout
+      - attach_workspace:
+          at: workspace
+      - run:
+          name: Gather and rename binaries from dependent jobs
+          command: |
+            mkdir github/
+            cp workspace/solc/solc             github/solc-static-linux
+            cp workspace/build/solc/solc       github/solc-macos
+            cp workspace/solc/Release/solc.exe github/solc-windows.exe
+            cp workspace/soljson.js            github/soljson.js
+      - store_artifacts:
+          path: github/
+      - run:
+          name: Rename binaries to solc-bin naming convention
+          command: |
+            full_version=$(
+              github/solc-static-linux --version |
+              sed -En 's/^Version: ([0-9.]+.*\+commit\.[0-9a-f]+(\.mod)?).*$/\1/p'
+            )
+
+            mkdir -p "solc-bin/{linux-amd64,macosx-amd64,windows-amd64,bin}"
+            mv github/solc-static-linux "solc-bin/linux-amd64/solc-linux-amd64-${full_version}"
+            mv github/macosx-amd64      "solc-bin/macosx-amd64/solc-macosx-amd64-${full_version}"
+            mv github/solc-windows.exe  "solc-bin/windows-amd64/solc-windows-amd64-${full_version}.exe"
+            mv github/soljson.js        "solc-bin/bin/soljson-${full_version}.js"
```

Comment:
This is something I could also put in a script (see `rename-circleci-binaries-for-solc-bin.sh` in #12181), which would allow running it locally instead of renaming manually (this workflow will become obsolete after this PR though). It also fits better with the idea of not keeping bigger pieces bash code outside of CI config.

The way it's done here is much simpler though.